### PR TITLE
fix(core/structure): include introductory sections in ToC by default

### PR DESCRIPTION
**Pull request by sidvishnoi**
Fixes https://github.com/w3c/respec/issues/3569
- Added a deprecation msg on use of `tocIntroductory`.
- `tocIntroductory: true` by default.

TODO:
- [x] Remove/deprecate from docs:
  - [x] https://respec.org/docs/#introductory
  - [x] https://respec.org/docs/#tocIntroductory
- [ ] Remove from ReSpec: `class="introductory"`, unless it has some special meaning outside ReSpec. It can lead of change in header IDs though.

### CODE REVIEW

**Comment by sidvishnoi** [comment]
Not as simple as I imagined it to be 😄

File: src/core/structure.js [hunk 1]
```diff
@@ -194,9 +192,12 @@ function filterHeader(h) {
 }
 
 export function run(conf) {
-  if ("tocIntroductory" in conf === false) {
-    conf.tocIntroductory = false;
+  if ("tocIntroductory" in conf) {
+    const msg = "Configuration option `tocIntroductory` is deprecated.";
+    const hint = `Add a 'notoc' class to remove a section from Table of Contents.`;
```
Comment (marcoscaceres) [inline src/core/structure.js:197]:
```suggestion
    const hint = `Add a 'notoc' class to remove a section from the Table of Contents.`;
```

**Review by marcoscaceres — APPROVED** [review]
Not sure about that test fail, but otherwise looks good!

File: src/core/structure.js [hunk 2]
```diff
@@ -194,9 +192,12 @@ function filterHeader(h) {
 }
 
 export function run(conf) {
-  if ("tocIntroductory" in conf === false) {
-    conf.tocIntroductory = false;
+  if ("tocIntroductory" in conf) {
```
Comment (marcoscaceres) [inline src/core/structure.js:195]: I think we should only warn when `tocIntroductory` is false... people doing `tocIntroductory: true` are already doing the right thing.  

```suggestion
  if ("tocIntroductory" in conf && tocIntroductory === "false") {
```

**Comment by marcoscaceres** [comment]
Updated docs: https://github.com/w3c/respec/wiki/tocIntroductory + https://github.com/w3c/respec-web-services/pull/211

**Comment by marcoscaceres** [comment]
@sidvishnoi, any timeframe on this one? Want me to take it over?

**Comment by sidvishnoi** [comment]
Feel free to take over. Need to identify why some tests are failing and update them (hopefully with minimal changes).

**Comment by marcoscaceres** [comment]
Firefox failures are very odd... can't reproduce locally :(

**Pull request merged**